zkevm-circuits/src/evm_circuit/execution/comparator.rs

```diff
@@ -0,0 +1,316 @@
+use crate::{
+    evm_circuit::{
+        execution::{
+            bus_mapping_tmp::{Block, Call, ExecStep, Transaction},
+            ExecutionGadget,
+        },
+        step::ExecutionState,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition, Transition::Delta,
+            },
+            from_bytes,
+            math_gadget::{ComparisonGadget, IsEqualGadget},
+            select, Word,
+        },
+    },
+    util::Expr,
+};
+use bus_mapping::{eth_types::ToLittleEndian, evm::OpcodeId};
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+// ComparatorGadget verifies ADD and SUB at the same time by an extra swap flag,
+// when it's ADD, we annotate stack as [a, b, ...] and [c, ...],
+// when it's SUB, we annotate stack as [a, c, ...] and [b, ...].
+// Then we verify if a + b is equal to c.
```

Comment:
Some incorrect parts in the comment for the gadget.

Reply:
Thanks for catching this, removed in [`665b9d1`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/665b9d10d3af47b6a1b6b7dbdcbb3158a070bf02)